src/data/custom_iconsets.ts

```diff
@@ -0,0 +1,15 @@
+export interface CustomIcons {
+  [key: string]: { path: string; viewBox?: string };
+}
+
+export interface CustomIconsetsWindow {
+  customIconsets?: Map<string, (name: string) => Promise<CustomIcons>>;
+}
+
+const customIconsetsWindow = window as CustomIconsetsWindow;
+
+if (!("customIconsets" in customIconsetsWindow)) {
+  customIconsetsWindow.customIconsets = new Map();
```

Comment:
Should this be defined by default?
It feels like a potential source of heissenbuggs when the load order changes.

Reply:
Custom icon sets will still need to check and define it if it is not there.

Reply:
Why?

Reply:
That's the thing. If it's not defined developers will soon discover  and learn that 
`window.customIconsets = window.customIconsets || {}` is required before using it.
If it's not, they'll ignore it and that will still work. Until it doesn't.

Reply:
I guess that is a question of documentation, it will only be defined when `ha-icon` is loaded, so yeah a developer should check if it is set.